drivers/gpio/Kconfig.rt1718s

```diff
@@ -0,0 +1,45 @@
+# RT1718S driver configuration options
+
+# Copyright 2022 Google LLC
+# SPDX-License-Identifier: Apache-2.0
+
+config GPIO_RT1718S
+	bool "RT1718S I2C-based TCPC chip with GPIOs"
+	default y
+	depends on DT_HAS_RICHTEK_RT1718S_GPIO_PORT_ENABLED
+	help
+	  Enable driver GPIO for RT1718S I2C-based TCPC chip.
+
+if GPIO_RT1718S
+
+config RT1718S_INIT_PRIORITY
+	int "RT1718S GPIO init priority"
+	default 30
+	help
+	  RT1718S device driver initialization priority. The priority should be
+	  lower than I2C device.
+
+config GPIO_RT1718S_PORT_INIT_PRIORITY
```

Comment:
I have would strongly suggest making this a devicetree property. What if on a system, there were too and one had to be started before the other?

Reply:
Is that possible? We have a single I2C_INIT_PRIORITY symbol.

Reply:
> Is that possible? We have a single I2C_INIT_PRIORITY symbol.

Yes, it's possible. 

> The initialization priorities are required to resolve to integer literals, and symbolic expressions are not permitted. 

DT properties *do* resolve to integer literals. 

> Potentially the correct DT_PROP usage could resolve correctly, but I don't see any prior art in Zephyr that uses devicetree for the initialization priority.

I wouldn't imagine that there would be prior art in Zephyr because most Zephyr devices don't have multiple off-chip GPIO controllers with a strong determinism requirement. My suggestion could also be solving a problem that nobody cares about (yet).

Feel free to ignore

Reply:
@cfriedt would it make sense to have a couple of extra common GPIO priority entries for off chip devices? Like GPIO_I2C_INIT_PRIORITY and GPIO_SPI_INIT_PRIORITY, and then have a couple of build checks to make sure that those are set correctly against I2C_INIT_PRIORITY and SPI_INIT_PRIORITY.